mautic/mautic
# Merge twitter monitor contacts based on name

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? |  X
| Related user documentation PR URL | Will come soon
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added a configurable option to Twitter social monitoring. When turned on Mautic will try to match people found on Twitter with already existing contacts based on the name of the twitter account. This avoids having people double in your contacts list (one raw, and one twitter) plus it's an extra feature to the social monitoring in general.

Hubspot has a similar feature. Do note that this is never 100% accurate, but it should be close enough to be very useful for most users. Hence the reason why I made it configurable.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
N/A

#### Steps to test this PR:
1. Create a social monitor (Twitter hashtag or mention)
2. Save
3. Set 'Match contact names' to 'Yes'
4. Start the Cronjob
5. See the magic happen

Note that in order for this to work you need a contact and a twitter account with the same name. i.e. contact name 'John Doe' and twitter account '@johndoe' with the name set as 'John Doe'. It will compare the contact name with the name set on twitter (so not the @ name).

#### List deprecations along with the new alternative:
N/A

#### List backwards compatibility breaks:
N/A
